move/interchain_token_service/sources/versioned/interchain_token_service_v0.move

```diff
@@ -215,7 +239,16 @@ module interchain_token_service::interchain_token_service_v0 {
     ): TokenId {
         let coin_info = coin_info::from_metadata_ref<T>(metadata);
 
-        self.register_coin(coin_info, coin_management, true)
+        let token_id = token_id::from_coin_data(
```

Comment:
i don't like the duplication. Introduce a new `register_coin_from_coin_info` that's deprecated, releases the metadata, and calls the new `register_coin` helper

Reply:
`register_coin` should be the function name in `interchain_token_service_v0` because, even though they are no longer versioned, that's what the `entry fun` is called in `interchain_token_service`. However, I'll remove the duplication